unified-runtime/examples/codegen/codegen.cpp

```diff
@@ -64,16 +64,21 @@ get_supported_adapters(std::vector<ur_adapter_handle_t> &adapters) {
 std::vector<ur_platform_handle_t>
 get_platforms(std::vector<ur_adapter_handle_t> &adapters) {
   uint32_t platformCount = 0;
-  ur_check(urPlatformGet(adapters.data(), adapters.size(), 1, nullptr,
-                         &platformCount));
+  std::vector<ur_platform_handle_t> platforms;
+  for (auto adapter : adapters) {
+    uint32_t adapterPlatformCount = 0;
+    urPlatformGet(adapter, 0, nullptr, &adapterPlatformCount);
 
+    platforms.reserve(platformCount + adapterPlatformCount);
+    urPlatformGet(adapter, adapterPlatformCount, &platforms[platformCount],
+                  &adapterPlatformCount);
+    platformCount += adapterPlatformCount;
+  }
   if (!platformCount) {
     throw std::runtime_error("No platforms available.");
   }
+  platforms.resize(platformCount);
```

Comment:
This is UB, the access to `platforms[platformCount]` is out of bounds, because `platforms.size()` is always 0 here. This should be using `resize()` and not `reserve()`.
In practice the later `.resize()` will reset all platforms to `nullptr` and make the example crash. See #18032